src/helpers.mjs

```diff
@@ -3,13 +3,14 @@ import { Type, Value, Descriptor } from './value.mjs';
 import { ToString, DefinePropertyOrThrow, CreateBuiltinFunction } from './abstract-ops/all.mjs';
 import { X, AwaitFulfilledFunctions } from './completion.mjs';
 
+const kNegativeZero = Symbol('kNegativeZero');
```

Comment:
You could make this `new Value(-0)` and then you don't need the new branches in the iterator methods.